### ISSUE TITLE
"There is no formatter for 'json'-files installed." error for JSON and TypeScript files

### ISSUE THREAD

**Issue by marcelh-gh** [Microsoft/vscode#51980]
- VSCode Version: 1.24.1
- OS Version: Win10 1803

It was working just fine with 1.24 .....
Uninstalled VS Code -> fresh install of 1.24.1 -> problem still exists.

Does this issue occur when all extensions are disabled?: Yes

**Comment by marcelh-gh** [comment]
this is not a duplicate of #46897 . It affects any size JSON and also TypeScript.

**Comment by aeschli** [comment]
Can you check whether `"json.format.enable": false` is set?

**Comment by aeschli** [comment]
Reasons for not getting the formatter:
- JSON extension disabled: Go to the extensions viewlet and enter `@disabled` in the search field. JSON language support should *not* show up
- `"json.format.enable": false`: Please set to `true`
- the file you have open is not associated with the JSON mode: Please verify that the editor shows 'JSON' in the lower right corner
- It's a huge file. Please also try with a small file to verify that this is the problem

**Comment by marcelh-gh** [comment]
@aeschli sorry for the late reply. I did some further testing and the conclusion is: this was **not** an issue of the 1.24.1 release. It actually **is** somewhat related to #46897.
I had a JSON of about 300KB size (not the hundreds of MB in #46897). If I used the JSON formatter on this file I would get the error message and subsequently it would not work on any JSON or TypeScript file. The only remedy: restart VS Code and then use the formatter inside a JavaScript file. after that it would work fine for JSON and TypeScript again.
